can/notifier.py

```diff
@@ -138,14 +143,22 @@ def _on_message_received(self, msg: Message) -> None:
                 self._loop.create_task(res)
 
     def _on_error(self, exc: Exception) -> bool:
-        listeners_with_on_error = [
-            listener for listener in self.listeners if hasattr(listener, "on_error")
-        ]
+        """Calls ``on_error()`` for all listeners if they implement it.
 
-        for listener in listeners_with_on_error:
-            listener.on_error(exc)
+        :returns: ``True`` if and only if at least one handler was called
```

Comment:
```suggestion
        :returns: ``True`` if at least one error handler was called.
```

I think it is easy to misread the "if and only if at least one ..." as "if and only if one ...".